sw/device/silicon_creator/rom/rom_isrs.S

```diff
@@ -25,15 +26,25 @@
   .balign 4
   .global rom_exception_handler
   .type rom_exception_handler, @function
+  .extern flash_ecc_exc_handler_en
 rom_exception_handler:
   // Save all registers to the exception frame.  The ROM locates its initial
   // stack_end at `ram_end - 128` and the stack grows downwards from there.
   // Save `sp` into `mscratch` and use the 128 bytes reserved at the top of
   // RAM as the exception frame.
+  csrw mscratch, sp
+
+  // Before saving the stack frame, check if handling of flash exceptions is
+  // enabled.  Determine this with just the stack pointer so that if not enabled,
+  // we don't have to do any other register save/restore.
+  la   sp, flash_ecc_exc_handler_en
+  lw   sp, 0(sp)
+  xori sp, sp, HARDENED_BOOL_TRUE
```

Comment:
For hardening, this will help the attacker bypass this check easier than to force it to be enabled. I think that is probably what we want in this case as disabling this handler via faults will not result in a weaker security.

Reply:
That's what I was thinking as well.